### DISCUSSION TITLE
How to deal with large functions and the 50MB size limit for e.g. Netlify or Vercel?

### DISCUSSION THREAD

**Discussion by ixam1** [remix-run/remix#4957] [General]
I am starting to try out Remix to build an app and loving it so far, but I cant deploy it to Netlify anymore, because I have reached the 50MB limit, this is what netlify says:

```
10:07:35 PM: 1 new functions to upload
10:07:41 PM: The function "SERVER" is larger than the 50MB limit. Please consider reducing it.
```

I am using firebase and some mail packages to handle recieving emails via IMAP and that already pushed me over 50MB. These are my current dependencies:

```
	"dependencies": {
		"@headlessui/react": "^1.7.7",
		"@heroicons/react": "^2.0.13",
		"@netlify/functions": "^1.3.0",
		"@remix-run/netlify": "^1.9.0",
		"@remix-run/node": "^1.9.0",
		"@remix-run/react": "^1.9.0",
		"clsx": "^1.2.1",
		"cross-env": "^7.0.3",
		"date-fns": "^2.29.3",
		"firebase": "^9.15.0",
		"firebase-admin": "^11.4.1",
		"imapflow": "^1.0.117",
		"mailparser": "^3.6.2",
		"react": "^18.2.0",
		"react-dom": "^18.2.0",
		"react-hot-toast": "^2.4.0",
		"reactfire": "^4.2.2",
		"usehooks-ts": "^2.9.1"
	},
```

As far as I have understood it, remix bundles basically the server into 1 netlify function. Which leads to the large size. Do you guys have any good suggestion to deal with that? It seems like this 50MB arent hard to reach.

I know I could deploy single netlify/firebase functions on my own. But that just is just so clunky in comparison to the awesome remix loaders/actions.

Is there any service that doesnt have a 50MB limit? I suppose deploy a node.js server, but wouldnt you loose all CDN/Edge advantages of Netlify/Vercel etc.?

**Comment by davidhousedev** [comment]
Are you querying for data stored in Firebase from your server? If so, you may find that you see faster request times by hosting your server on Firebase, compared to Netlify or Cloudflare.

While it's true that your compute could be located further away from your users, it also means that your application could retrieve data very easily because it the request would stay within the data center.

Your mileage may vary, just something to consider :)

**Reply by ixam1** [reply]
Huh good point I havent even thought about hosting on firebase, that actually makes a lot of sense.

Seems like firebase also has 10 times the limit (500MB): https://firebase.google.com/docs/functions/quotas#:~:text=100MB%20(compressed)%20for%20sources.%0A500MB%20(uncompressed)%20for%20sources%20plus%20modules.

Thanks for the suggestion, I will try that!
